control-admin/src/main/java/fi/nls/oskari/control/admin/LayerAdminHandler.java

```diff
@@ -123,6 +140,32 @@ public void handleDelete(ActionParameters params) throws ActionException {
         }
     }
 
+    public List<OskariLayer> cleanupLayerReferences(int layerId) throws ActionException {
+        List<OskariLayer> layers = mapLayerService.findAll();
```

Comment:
There's duplicated code here from the usage check handler. I'm wondering if we should have a helper that could be used in both. Just report report usage on the other handler and LayerAdminHandler could use that same helper to get a list of layers with reference the input layer id. Thoughts on this?

Reply:
I mean it's not bad but with nothing to tie these two together it's pretty easy to miss updating the logic in one when updating it on the other one.

Reply:
Agreed. It's better to do that layer looping only once in one place. I'll check this.